gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -427,9 +453,88 @@ public void delete(URI path, boolean recursive) throws IOException {
     repairImplicitDirectory(parentInfoFuture);
   }
 
-  /** Deletes all items in the given path list followed by all bucket items. */
+  /**
+   * Helper function to check if the folderResource is a directory or bucket.
+   *
+   * @param bucketName
+   * @param folderName
+   * @return true if directory else false
+   */
+  private boolean isFolderResource(String bucketName, String folderName) {
+    return !Strings.isNullOrEmpty(bucketName) && !Strings.isNullOrEmpty(folderName);
+  }
+
+  /**
+   * Return the bucket name if exists else return empty string
+   *
+   * @param path
+   * @return bucket name
+   */
+  private String getBucketName(@Nonnull URI path) {
+    return (Strings.isNullOrEmpty(path.getAuthority())) ? "" : path.getAuthority();
```

Comment:
We do not expect authority to be null, correct?

Reply:
NO, modified by adding checkState just for sanity check